components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt

```diff
@@ -593,33 +616,39 @@ class QrFragment : Fragment() {
         @Volatile internal var qrState: Int = 0
     }
 
-    internal class AsyncScanningTask(
-        private val scanCompleteListener: OnScanCompleteListener?,
-        private val multiFormatReader: MultiFormatReader = MultiFormatReader()
-    ) : AsyncTask<BinaryBitmap, Void, Void>() {
-
-        override fun doInBackground(vararg bitmaps: BinaryBitmap): Void? {
-            return processImage(bitmaps[0])
+    @VisibleForTesting
+    internal suspend fun tryScanningBitmap(bitmap: BinaryBitmap, scanCompleteListener: OnScanCompleteListener?) {
+        scanCompleteListener?.let {
+            withContext(Dispatchers.Default) {
```

Comment:
We're switching context here and then again in `decodeBitmap`. Looks like we can skip it here?

Reply:
I thought of decoupling them, make sure that anything `tryScanningBitmap` will call will run on a background thread.
But indeed, probably yagni. Thanks!

Reply:
> I thought of decoupling them, make sure that anything tryScanningBitmap will call will run on a background thread.

OK, got it. Did you see my comment below re: "background thread". It seems right now we run this on the main thread?

Reply:
Previously we would just call `tryScanningBitmap` from Main but it would then run on a background thread because of `withContext(Dispatchers.Default)`.
To simplify things now I call `tryScanningBitmap` straight from `Dispatchers.Default` and continue execution in the background until `onScanComplete` posts to Main in case of a successfully identified QR code.